File: 3658-gcd-of-odd-and-even-sums/3658-gcd-of-odd-and-even-sums.cpp
class Solution {
public:
    int gcd(int a,int b){
            while (b!=0){
                int temp = b;
                b = a%b;
                a = temp;
            }
            return a;
        }
    int gcdOfOddEvenSums(int n) {
        int x = n+n;
        int even=0;
        int odd =0;
        for (int i=1;i<x+1;i++){
            if (i%2==0){
                even+=i;
            }else{
                odd+=i;
            }
        }
        return gcd(odd,even);
    }
};

File: 3658-gcd-of-odd-and-even-sums/3658-gcd-of-odd-and-even-sums.py
class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        a=n+n
        even=0
        odd=0
        for i in range(1,a+1):
            if i%2==0:
                even+=i 
            else:
                odd+=i 
        def gcd(a,b):
            while (b!=0):
                a,b = b,a%b 
            return a
        return gcd(odd,even)
        

        